python/cuml/ensemble/randomforestregressor.pyx

```diff
@@ -496,16 +490,20 @@ class RandomForestRegressor(Base):
     def _tl_model_handles(self, model_bytes):
         task_category = REGRESSION_MODEL
         cdef ModelHandle tl_model_ptr = NULL
+        cdef uintptr_t tl_handle_int
+        cdef ModelHandle tl_handle = NULL
         cdef RandomForestMetaData[float, float] *rf_forest = \
             <RandomForestMetaData[float, float]*><uintptr_t> self.rf_forest
-        build_treelite_forest(& tl_model_ptr,
-                              rf_forest,
-                              <int> self.n_cols,
-                              <int> task_category,
-                              <vector[unsigned char] &> model_bytes)
-        mod_handle = <uintptr_t> tl_model_ptr
-
-        return ctypes.c_void_p(mod_handle).value
+        if model_bytes:  # bytes -> Treelite
```

Comment:
The if else statement here is similar to the one used in `_obtain_treelite_handle` function. I t would be cool if we can use the `_obtain_treelite_handle` func here in place of the if else statement

Reply:
`_tl_model_handles()` is not exactly identical to `_obtain_treelite_handle`, because `_tl_model_handles()` is converting incoming bytes (`model_bytes`) into a Treelite handle.

`_tl_model_handles()` is actually used in only one place:
https://github.com/rapidsai/cuml/blob/b9cb5c6c6da39ca6b0cb910356886692a4bcbddb/python/cuml/dask/ensemble/base.py#L103-L104
where `model_bytes` are assumed to be valid. So we can simplify this function by eliminating the else condition. 

I have clarified the intent of this function by renaming it to `_tl_handle_from_bytes()` and requiring a non-empty `model_bytes` argument.